pkg/builder/options.go

```diff
@@ -107,6 +107,20 @@ var (
 	// metav1.PartialObjectMetadata to the client when fetching objects in your
 	// reconciler, otherwise you'll end up with a duplicate structured or
 	// unstructured cache.
+	//
+	// When watching a resource with OnlyMetadata, for example the v1.Pod, you
+	// should not Get and List using the v1.Pod type. Instead, you should use
+	// the special metav1.PartialObjectMetadata type:
+	//
+	//   pod := &v1.Pod{}                             // ❌
+	//   mgr.GetClient().Get(ctx, nsAndName, pod)     // ❌
+	//
+	//   pod := &metav1.PartialObjectMetadata{}       // ✅
```

Comment:
Doesn't this need typemeta set

Reply:
That's a very good point, controller-runtime would have a hard time figuring out which type we are referring to with just `&metav1.PartialObjectMetadata{}` 😅

I'll fix this.